dataprofiler/tests/data_readers/test_csv_graph_data.py

```diff
@@ -0,0 +1,142 @@
+import os
+import unittest
+from io import BytesIO, StringIO, TextIOWrapper
+
+from dataprofiler.data_readers.data_utils import is_stream_buffer
+from dataprofiler.data_readers.graph_data import GraphData
+
+test_root_path = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+
+
+class TestGraphDataClass(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        
+        test_dir = os.path.join(test_root_path, 'data')
+        cls.input_file_names = [
+            dict(path=os.path.join(test_dir, 'csv/graph-differentiator-input-positive.csv'),
+                 count=7, delimiter=',', has_header=[0],
+                 num_columns=4, encoding='utf-8'),
+            dict(path=os.path.join(test_dir, 'csv/graph-differentiator-input-standard-positive.csv'),
+                 count=7, delimiter=',', has_header=[0],
+                 num_columns=11, encoding='utf-8'),
+            dict(path=os.path.join(test_dir, 'csv/graph-differentiator-input-negative.csv'),
+                 count=4, delimiter=',', has_header=[0],
+                 num_columns=4, encoding='utf-8'),
+            dict(path=os.path.join(test_dir, 'csv/graph-data-input-json.json'),
+                 count=4, delimiter=',', has_header=[0],
+                 num_columns=4, encoding='utf-8')
+        ]
+
+        cls.buffer_list = []
+        for input_file in cls.input_file_names:
+            # add StringIO
+            buffer_info = input_file.copy()
+            with open(input_file['path'], 'r', encoding=input_file['encoding']) as fp:
+                buffer_info['path'] = StringIO(fp.read())
+            cls.buffer_list.append(buffer_info)
+            
+            # add BytesIO
+            buffer_info = input_file.copy()
+            with open(input_file['path'], 'rb') as fp:
+                buffer_info['path'] = BytesIO(fp.read())
+            cls.buffer_list.append(buffer_info)
+
+        cls.file_or_buf_list = cls.input_file_names + cls.buffer_list
```

Comment:
use this for the `is_match` for loop similar to https://github.com/capitalone/DataProfiler/blob/main/dataprofiler/tests/data_readers/test_csv_data.py#L461